version: "3"

services:
  backend:
    build: ./back-end
    restart: always
    working_dir: /app
    volumes:
      - ./back-end/src:/app
      - ./back-end/resources:/resources
    command: python3.8 san11_platform_server.py
    depends_on:
      - db

  envoy:
    image: envoyproxy/envoy-alpine:v1.13.0
    volumes:
      - ./back-end/envoy.yaml:/etc/envoy/envoy.yaml
    command: /usr/local/bin/envoy -c /etc/envoy/envoy.yaml -l trace --log-path /tmp/envoy_info.log
    ports:
      - "8090:8090"

  db:
    build: ./back-end/database
    restart: always
    volumes: 
      - ./back-end/database/postgres-data:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: example
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

  frontend:
    build: ./front-end
    working_dir: /app
    volumes:
      - ./front-end:/app
    command: bash -c "npm install && ng serve --host 0.0.0.0 --disableHostCheck"
    depends_on: 
      - backend
    ports:
      - "80:4200"
    
